public/app/plugins/datasource/cloud-monitoring/datasource.ts

```diff
@@ -134,10 +106,19 @@ export default class CloudMonitoringDatasource extends DataSourceApi<CloudMonito
       .toPromise();
   }
 
+  applyTemplateVariables(target: CloudMonitoringQuery, scopedVars: ScopedVars): Record<string, any> {
```

Comment:
This function is a really nice addition to the api. However, I don't think it does what its name stipulates - the template variables seem to be applied in the prepareTimeSeriesQuery function. Is it possible to remove that function and put all that logic in applyTemplateVariables instead?

Reply:
this is a bit difficult, since the function is also used in interpolateVariablesInQueries, i don't know whether it would break something if i changed the call to preparetimeseriesquery to the call to applyTemplateVariables